### PR TITLE
Adds cake cache clear_group command

### PR DESCRIPTION
**Pull request by cnizzardini**
Adds `bin/cake cache clear_group {groupname}` command. This is handy for clearing cache in mass as part of deployments or specific groups as part of cronjobs (for me atleast). If this is acceptable I'll add tests and documentation.

### CODE REVIEW

**Review by markstory — COMMENTED** [review]
I think this is a useful command line tool to have. Could we target 4.next with the pull request though?

File: src/Command/CacheClearGroupCommand.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     * @param \Cake\Console\ConsoleOptionParser $parser The parser to be defined
+     * @return \Cake\Console\ConsoleOptionParser The built parser.
+     */
+    public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        $parser = parent::buildOptionParser($parser);
+        $parser->setDescription('Clear all data in a single cache group.');
+        $parser->addArgument('group', [
+            'help' => 'The cache group to clear. For example, `cake cache clear_group mygroup` will clear ' .
+                'all caches belonging to group "mygroup".',
```
Comment (markstory) [inline src/Command/CacheClearGroupCommand.php]:
```suggestion
        $parser->addArgument('group', [
            'help' => 'The cache group to clear. For example, `cake cache clear_group mygroup` will clear ' .
                'all cache items belonging to group "mygroup".',
```

**Comment by LordSimal** [comment]
LGTM 👍🏻

File: src/Command/CacheClearGroupCommand.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            $io->error(sprintf('Cache config "%s" not found', $config));
+
+            return static::CODE_ERROR;
+        }
+
+        if (!Cache::clearGroup($group, $args->getArgument('config'))) {
+            $io->error(sprintf('Error encountered clearing group "%s"', $group));
+
+            return static::CODE_ERROR;
+        }
```
Comment (markstory) [inline src/Command/CacheClearGroupCommand.php]: Is the goal of the command to clear the *entire* group of cache keys across all configurations? Or is it more a direct wrapper around `Cache::clearGroup()` which counterintuitively only clears a group for a single cache profile.

Should we perhaps clear *all* cache profiles if no `--config` option is provided? If one is provided we could clear only the profile that was named.

Comment (cnizzardini) [reply src/Command/CacheClearGroupCommand.php]: > Should we perhaps clear all cache profiles if no --config option is provided?

Agreed.

Comment (cnizzardini) [reply src/Command/CacheClearGroupCommand.php]: How should the command respond in the unlikely event that one cache is cleared and the other isn't? It looks like `clear_all` will abort at the first sign of trouble. Is that what should be done here as well @markstory?

Comment (markstory) [reply src/Command/CacheClearGroupCommand.php]: Yeah, I think we can only do a best effort, and report the first failure.

File: src/Command/CacheClearGroupCommand.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        $parser = parent::buildOptionParser($parser);
+        $parser->setDescription('Clear all data in a single cache group.');
+        $parser->addArgument('group', [
+            'help' => 'The cache group to clear. For example, `cake cache clear_group mygroup` will clear ' .
+                'all cache items belonging to group "mygroup".',
+            'required' => true,
+        ]);
+        $parser->addArgument('config', [
+            'help' => 'Name of the configuration to use. Defaults to "default".',
```
Comment (markstory) [inline src/Command/CacheClearGroupCommand.php]:
```suggestion
            'help' => 'Name of the configuration to use. Defaults to no value which clears all cache configurations.',
```

File: src/Command/CacheClearGroupCommand.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            return static::CODE_ERROR;
+        }
+
+        foreach ($groupConfigs[$group] as $groupConfig) {
+            if ($config !== null && $config !== $groupConfig) {
+                continue;
+            }
+
+            if (!Cache::clearGroup($group, $groupConfig)) {
+                $io->error(sprintf('Error encountered clearing group "%s"', $group));
```
Comment (markstory) [inline src/Command/CacheClearGroupCommand.php]:
```suggestion
                $io->error(sprintf('Error encountered clearing group "%s". Was unable to clear entries for "%s".', $group, $groupConfig));
```

File: src/Command/CacheClearGroupCommand.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                $this->abort();
+            } else {
+                $io->success(sprintf('Group "%s" was cleared', $group));
```
Comment (markstory) [inline src/Command/CacheClearGroupCommand.php]:
```suggestion
                $io->success(sprintf('Group "%s" was cleared.', $group));
```

**Comment by markstory** [comment]
Looking good. There is one code style issue remaining though.

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**